src/components/ClaimENSPage/ClaimENSPage.tsx

```diff
@@ -1,16 +1,62 @@
 import * as React from 'react'
-import { Row, Column, Section, Narrow, InputOnChangeData, Header, Form, Field, Button, Mana } from 'decentraland-ui'
+import { Address } from 'web3x-es/address'
+import { Row, Column, Section, Narrow, InputOnChangeData, Header, Form, Field, Button, Mana, Radio } from 'decentraland-ui'
 import { t } from 'decentraland-dapps/dist/modules/translation/utils'
 import { locations } from 'routing/locations'
 import Back from 'components/Back'
 import LoggedInDetailPage from 'components/LoggedInDetailPage'
 import { MAX_NAME_SIZE, isNameValid, PRICE } from 'modules/ens/utils'
 import { Props, State } from './ClaimENSPage.types'
+import { createEth } from 'decentraland-dapps/dist/lib/eth'
+
 import './ClaimENSPage.css'
+import { ERC20 as MANAToken } from 'contracts/ERC20'
+import { CONTROLER_ADDRESS, MANA_ADDRESS } from 'modules/common/contracts'
+import { Eth } from 'web3x-es/eth'
 
 export default class ClaimENSPage extends React.PureComponent<Props, State> {
   state: State = {
-    name: ''
+    name: '',
+    amountApproved: 0,
+    isLoading: false,
+    tx: undefined
+  }
+
+  async componentDidUpdate() {
+    const eth: Eth | null = await createEth()
+    const { address } = this.props
+
+    if (eth && address) {
+      const contractMANA = new MANAToken(eth, Address.fromString(MANA_ADDRESS))
+      const allowance: string = await contractMANA.methods
+        .allowance(Address.fromString(address), Address.fromString(CONTROLER_ADDRESS))
+        .call()
+      const amountApproved: number = +allowance
+      this.setState({ amountApproved })
+    }
+    const { tx } = this.state
+    if (tx && tx.status) {
+      this.setState({ isLoading: false, tx: undefined })
+    }
+  }
+
+  handleManaApprove = async () => {
+    const { address } = this.props
+    const eth: Eth | null = await createEth()
+    if (!eth) return
```

Comment:
This will silently fail...when would this happen? should we handle it?
@cazala

Reply:
mm currently this would only happen if `window.ethereum` doesn't exist, like when the user doesn't have metamask or in incognito mode, so i think i could never happen at this point since you have to login before you can get here, right?

Reply:
Right, ok!